tests/test_pipe.py

```diff
@@ -7,12 +8,41 @@ def test_pipe_initialized():
     nums = [1, 2, 3, 4]
     fn = lambda x: x * 2
     pipe = Pipe()
+    map_task1 = MapTask(fn=fn)
+    map_task2 = MapTask(fn=fn)
 
     stage = (
         pipe.set_source(LocalSource(data=nums))
-        .add_task(MapTask(fn=fn))
-        .add_task(MapTask(fn=fn))
+        .add_task(map_task1)
+        .add_task(map_task2)
         .run()
     )
 
-    stage.result == [fn(num) for num in nums]
+    assert stage.result == [fn(fn(num)) for num in nums]
+    assert stage.status == StageStatus.DONE
+    assert stage.task == map_task2
+
+
+def test_pipe_exception_occurred(capsys):
+    nums = [1, 2, 3, 4]
+
+    def except_occurred(_):
+        raise Exception("error")
+
+    except_task = MapTask(fn=except_occurred, fallback=lambda x: print(x))
+    map_task = MapTask(fn=lambda x: x * 2)
+    pipe = Pipe()
+
+    stage = (
+        pipe.set_source(LocalSource(data=nums))
+        .add_task(except_task)
+        .add_task(map_task)
+        .run()
+    )
+
+    assert stage.status == StageStatus.ERROR
+    assert stage.result == []
+    assert stage.task == except_task
```

Comment:
I cannot find any part that update task field in stage class. Does this test pass?

Reply:
it works well.
attached the code below.

https://github.com/Giving-Street/pyper/blob/267496993c2227953c91da652171de8018cfe167/pyper/core/pipe.py#L48

Reply:
it works well.
attached the code below.

https://github.com/Giving-Street/pyper/blob/267496993c2227953c91da652171de8018cfe167/pyper/core/pipe.py#L48